### ISSUE TITLE
Haven advisers cause missions to crash

### ISSUE THREAD

**Issue by jmartinez989** [long-war-2/lwotc#147]
So this is a very weird one and admittedly I am only opening this one up for now just to make sure that we remember this is a thing but there seems to be a problem with soldiers being haven advisers.

If a soldier is a haven adviser there seems to be an issue if they partake in a job related retaliation event (ie. intel raid, supply raid, recruit raid). So the first one of these you can take an adviser on and it seems to be no problem. After that first one though the game seems to put those advisers in some sort of bad state as after that first event they will no longer be able to go on any other mission as they will cause the game to crash. I have tested this by taking a soldier that was a haven adviser at one point and had participated in a mini retaliation. After that event I tried to take them on another mission and the game crashed every single time., even if they are the only soldier on the mission and even if I only have lwotc and highlander running.

This is hard to provide a save for because of how you have to take them on a specific type of mission first.

As far as rendezvous go there seems to be no problem there, you can use them for any number of rendezvous but once you take them on one of those mini retals they just cannot go on any other mission.

**Comment by pledbrook** [comment]
I think I just had a Rendezvous crash. Unfortunately was away from computer when it happened and then restarted before checking the log. No idea if it was adviser related or not.

**Comment by jmartinez989** [comment]
yeah I might be. If that adviser was on a retal mission before then chances are they will likely crash on all future missions.

**Comment by pledbrook** [comment]
False alarm. All missions are crashing.

**Comment by jmartinez989** [comment]
So I tried to test this and my theory does not seem to hold water. Just took a soldier that was part of a mini retaliation (an intel one) and they did not crash the game.

Though I did notice something odd about it. The soldiers loadout did not apply. The only thing they loaded in with was their Weapon. They did not have any of the utility items I gave them and while they were wearing plated armor they were not benefiting from the health  bonuses that the armor provides.

Definitely something weird going on with haven advisers and mini retals.

**Comment by jmartinez989** [comment]
So I am going to close this one. Does not seem to be happening anymore and could have just been a fluke before. Will keep my eye on it and try to have a save of a campaign with this.

**Comment by jmartinez989** [comment]
Opening this again. So it looks like it popped up again. Haven adviser took part in a full retal (the vanilla kind where you load in and all of the rebels are spread out and you have to find them one by one). After I finished that one an intel raid popped up in the same region and I tried to do that one but every time I load into it I crash. Doesn't matter what mods I have or who I send in. Thing is if you could somehow take that soldier to any other mission they game will crash guaranteed.

I'll pull up a save file of the retal being available just before launch.

**Comment by jmartinez989** [comment]
[RetalHavenAdviserCrash.zip](https://github.com/long-war-2/lwotc/files/3167942/RetalHavenAdviserCrash.zip)

**Comment by jmartinez989** [comment]
So I think I have figured out what is going on. I think this only happens if your haven adviser takes damage when they are on a retaliation. On the last retaliation I did the haven adviser took damage but then after the mission the UI states they are fine. I think what happens here is that the game puts the adviser in a weird state of being wounded and fine at the same time. What seems to clear this is if after the retaliation you take them out of the haven and replace them with someone else.
